packages/block-editor/src/components/grid-interactivity/grid-item-pin-toolbar-item.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { ToolbarGroup, ToolbarButton } from '@wordpress/components';
+import { pin as pinIcon } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import BlockControls from '../block-controls';
+import { __unstableUseBlockElement as useBlockElement } from '../block-list/use-block-props/use-block-refs';
+import { getComputedCSS, getGridLines, getClosestLine } from './utils';
+
+export function GridItemPinToolbarItem( { clientId, layout, onChange } ) {
+	const blockElement = useBlockElement( clientId );
+	if ( ! blockElement ) {
+		return null;
+	}
+
+	const isPinned = !! layout?.columnStart && !! layout?.rowStart;
+
+	function unpinBlock() {
+		onChange( {
+			columnStart: undefined,
+			rowStart: undefined,
+		} );
+	}
+
+	function pinBlock() {
```

Comment:
I'm out of the loop so ready to be schooled on this, but, what exactly is the value of the "pinned" state? It's not exactly clear to me.

Is it to _indicate_ that a block has a col/row value? Or, to the user, that a block resides in a col/row that is not the start?


https://github.com/WordPress/gutenberg/assets/6458278/4a072850-e712-4866-9775-fd114baf150a

If so, should he pinned state be reset if I drag my block back to the first cell?

Reply:
The hypothesis is that it would be helpful to distinguish between items that are arranged automatically by the browser versus items that are going to appear in a specific cell because the user set it that way. Otherwise when other content is added the user might not be clear why some blocks move around but other blocks don't.

So we thought we'd try "pinned" as a metaphor for denoting which blocks won't automatically move around. I suspect this might be utterly confusing to users and that we'll have to come up with a better metaphor or interface. But good to get feedback first.

Reply:
Thinking out loud. (I'm in a typing mood.) A direction I'd like to explore is:

- when the grid is in Auto mode, all grid items are automatically placed by the browser.
- when the grid is in Manual mode, all grid items are manually placed by the user.

It's tricky though because there's lots of ways to add a block to a container that don't involve indicating where you want the block to go, e.g. the default block appender, the inserter, the _Move to_ command. Do we disable the block appender and inserter for manual Grids and force users to select a cell? Much to think about.

For now I think pinned / unpinned is okay. Things can be clunky while behind the experimental flag.

Reply:
What benefit would having to place items in the manual grid manually bring? I think it's nice to have the ability to do that (e.g. create a 6x6 empty grid and populate it with blocks in whatever pattern) but not sure there's any advantage to _forcing_ that behaviour in manual mode.

Reply:
I'm basically just interested in seeing if creating a really strong distinction between the two modes means that the feature as a whole is more understandable by non technical users. It's something we'll have to test in the field though, I don't think we have the necessary data to decide right now 😀